src/hydra-eval-jobs/hydra-eval-jobs.cc

```diff
@@ -442,45 +442,66 @@ int main(int argc, char * * argv)
         for (auto i = state->jobs.begin(); i != state->jobs.end(); ++i) {
             auto jobName = i.key();
             auto & job = i.value();
+            // For the error message
+            std::string lastTriedJobName = i.key();
 
             auto named = job.find("namedConstituents");
             if (named == job.end()) continue;
 
-            if (myArgs.dryRun) {
-                for (std::string jobName2 : *named) {
-                    auto job2 = state->jobs.find(jobName2);
-                    if (job2 == state->jobs.end())
-                        throw Error("aggregate job '%s' references non-existent job '%s'", jobName, jobName2);
-                    std::string drvPath2 = (*job2)["drvPath"];
-                    job["constituents"].push_back(drvPath2);
-                }
-            } else {
-                auto drvPath = store->parseStorePath((std::string) job["drvPath"]);
-                auto drv = store->readDerivation(drvPath);
-
-                for (std::string jobName2 : *named) {
-                    auto job2 = state->jobs.find(jobName2);
-                    if (job2 == state->jobs.end())
-                        throw Error("aggregate job '%s' references non-existent job '%s'", jobName, jobName2);
-                    auto drvPath2 = store->parseStorePath((std::string) (*job2)["drvPath"]);
-                    auto drv2 = store->readDerivation(drvPath2);
-                    job["constituents"].push_back(store->printStorePath(drvPath2));
-                    drv.inputDrvs[drvPath2] = {drv2.outputs.begin()->first};
-                }
+            try {
+                if (myArgs.dryRun) {
+                    for (std::string jobName2 : *named) {
+                        lastTriedJobName = jobName2;
+                        auto job2 = state->jobs.find(jobName2);
+                        if (job2 == state->jobs.end())
+                            throw Error("aggregate job '%s' references non-existent job '%s'", jobName, jobName2);
+                        std::string drvPath2 = (*job2)["drvPath"];
+                        job["constituents"].push_back(drvPath2);
+                    }
+                } else {
+                    auto drvPath = store->parseStorePath((std::string) job["drvPath"]);
+                    auto drv = store->readDerivation(drvPath);
+
+                    for (std::string jobName2 : *named) {
+                        lastTriedJobName = jobName2;
+                        auto job2 = state->jobs.find(jobName2);
+                        if (job2 == state->jobs.end())
+                            throw Error("aggregate job '%s' references non-existent job '%s'", jobName, jobName2);
+
+                        if ((*job2).find("error") != (*job2).end()) {
```

Comment:
Now that tripped me up. I get that this comes from `auto job2` being the result of the search. Is there a pattern we could be using so that we don't have to dereference all the time?

Reply:
`auto& something = *job2` should do the trick here.

Reply:
Honestly, this is _probably_ fine? I'd probably replace the `(*job2).` with just `job2->` but other than that this seems... fine.